# **1.0 Notebook of Initial Exploration and Analysis of the Training Data**
by jack phelan ^_^

## **EDA Report Outline/Checklist**

### Dataset overview
- Dataset source, number of records (rows) and variables (columns), file format, any metadata or documentation available
### Data quality summary
- Summarize the quality and completeness of the raw data
- Identify unusual data points or distributions (outliers)
- Statistical summary of dataset
- Summary statistics for measures (mean, median, std dev) and frequency distributions (counts) for categories 
### Univariate analysis
- Visualizations of each variable (histograms or bar charts).  Can be done with Tableau or Python
- Annotate any significant observations
### Bivariate analysis
- Bivariate visualizations of each variable vs the response variable and any other “interesting” pairs of variables. Can be done with Tableau or Python
Annotate any significant observations
    

In [ ]:
#imports
import pandas as pd 
import numpy as np


In [4]:
# ingest from file
df = pd.read_csv('../data/healthcare_readmissions_dataset_train.csv')

display(df.head())

,PatientID,Age,Gender,Ethnicity,Hospital ID,Height (m),Smoker,BMI,Weight (kg),Adjusted Weight (kg),Has Diabetes,Has Hypertension,Exercise Frequency,Diet Type,Number of Prior Visits,Medications Prescribed,Length of Stay,Type of Treatment,Readmission within 30 Days
0,1000000,23,Female,African American,Hosp2,1.6,False,25.0,64.0,63.283346,0,0,Regular,High-fat,3.0,3.0,0,NaN,0
1,1000002,56,Female,Hispanic,Hosp3,1.8,True,27.0,87.5,87.678859,0,0,Regular,High-fat,2.0,NaN,2,NaN,0
2,1000003,28,Male,African American,Hosp1,1.8,False,35.0,113.4,113.497844,0,1,NaN,Other,NaN,2.0,5,NaN,0
3,1000004,70,Female,Caucasian,Hosp2,1.8,False,27.7,89.7,89.717694,0,0,NaN,Other,3.0,NaN,0,Major Surgery,0
4,1000005,48,Female,Hispanic,Hosp1,1.9,False,22.4,80.9,80.528927,0,0,Occasional,High-fat,7.0,5.0,7,Major Surgery,1


In [12]:
print(df.shape[0],"Rows", df.shape[1], "Columns")

8038 Rows 19 Columns


In [13]:
display(df.info())

<class 'pandas.DataFrame'>
RangeIndex: 8038 entries, 0 to 8037
Data columns (total 19 columns):
 #   Column                      Non-Null Count  Dtype  
---  ------                      --------------  -----  
 0   PatientID                   8038 non-null   int64  
 1   Age                         8038 non-null   int64  
 2   Gender                      8038 non-null   str    
 3   Ethnicity                   8038 non-null   str    
 4   Hospital ID                 8038 non-null   str    
 5   Height (m)                  8038 non-null   float64
 6   Smoker                      8038 non-null   bool   
 7   BMI                         8038 non-null   float64
 8   Weight (kg)                 8038 non-null   float64
 9   Adjusted Weight (kg)        8038 non-null   float64
 10  Has Diabetes                8038 non-null   int64  
 11  Has Hypertension            8038 non-null   int64  
 12  Exercise Frequency          5688 non-null   str    
 13  Diet Type                   8038 non-null   

None

### **Initial Notes on Columns**
- **PatientID**: Unique identifier/tracker; not a feature to use for modeling. 
- **Hospital ID**: Look into it further
- **Height**,**Weight**, and  **BMI** are likely all correlated
- Look into differences between **Weight** and **Adjusted Weight**
- Unify formatting for all columns : )
### **Column Type Breakdown**
- **Measures**: Height, Weight, Adjusted Weight, BMI
- **Booleans**: Gender(?), Smoker, Has Diabetes, Has Hypertension
- **Categorical**: Hospital ID, Excercise Frequency, Diet Type, Numbeer of Prior Visits, Medications Prescribed, Length of Stay
### **Missingness Observations**
- Exercise Frequency: 5688 rows (29.2% missing)
- Number of Prior Visits: 7724 rows (3.9% missing)
- Medications Prescribed: 7381 rows (8.1% missing)
- Type of Treatment: 5552 rows (30.9% missing)

In [ ]:
# formatting changes

df = df.copy()

